controllers/renovate_util.go

```diff
@@ -311,3 +313,146 @@ func CreateRenovaterJob(ctx context.Context, client client.Client, scheme *runti
+// - Users can view the results of pipelines and the results are stored, making debugging much easier
+// - Tekton automatically provides docker config from linked service accounts for private images, with a job I would need to implement this manually
+//
+// Warning: the installation token used here should only be scoped to the individual repositories being updated
+func CreateRenovaterPipeline(ctx context.Context, client client.Client, scheme *runtime.Scheme, namespace string, installations []installationStruct, slug string, debug bool, js func(slug string, repositories []renovateRepository, info interface{}) (string, error), info interface{}) error {
+	log := logger.FromContext(ctx)
+	log.Info(fmt.Sprintf("Creating renovate pipeline for %d installations", len(installations)))
+
+	if len(installations) == 0 {
+		return nil
+	}
+	timestamp := time.Now().Unix()
+	name := fmt.Sprintf("renovate-pipeline-%d-%s", timestamp, getRandomString(5))
+	secretTokens := map[string]string{}
+	configmaps := map[string]string{}
+	renovateCmds := []string{}
+	for _, installation := range installations {
+		secretTokens[fmt.Sprint(installation.id)] = installation.token
+		config, err := js(slug, installation.repositories, info)
+		if err != nil {
+			return err
+		}
+		configmaps[fmt.Sprintf("%d.js", installation.id)] = config
+
+		log.Info(fmt.Sprintf("Creating renovate config map entry for %d installation with length %d and value %s", installation.id, len(config), config))
+		renovateCmds = append(renovateCmds,
+			fmt.Sprintf("RENOVATE_TOKEN=$TOKEN_%d RENOVATE_CONFIG_FILE=/configs/%d.js renovate", installation.id, installation.id),
+		)
+	}
+	if len(renovateCmds) == 0 {
+		return nil
+	}
+	secret := &v1.Secret{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		StringData: secretTokens,
+	}
+	configMap := &v1.ConfigMap{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: configmaps,
+	}
+	trueBool := true
+	falseBool := false
+	renovateImageUrl := os.Getenv(RenovateImageEnvName)
+	if renovateImageUrl == "" {
+		renovateImageUrl = DefaultRenovateImageUrl
+	}
+	pipelineRun := &tektonapi.PipelineRun{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: tektonapi.PipelineRunSpec{
+			PipelineSpec: &tektonapi.PipelineSpec{
+				Tasks: []tektonapi.PipelineTask{{
+					Name: "renovate",
+					TaskSpec: &tektonapi.EmbeddedTask{
+						TaskSpec: tektonapi.TaskSpec{
+							Steps: []tektonapi.Step{{
+								Name:  "renovate",
+								Image: renovateImageUrl,
+								EnvFrom: []v1.EnvFromSource{
+									{
+										Prefix: "TOKEN_",
+										SecretRef: &v1.SecretEnvSource{
+											LocalObjectReference: v1.LocalObjectReference{
+												Name: name,
+											},
+										},
+									},
+								},
+								Command: []string{"bash", "-c", strings.Join(renovateCmds, "; ")},
```

Comment:
Does joining multiple `renovate` commands with `;` potentially hide failures?

Reply:
This is how the existing renovate job works. Also even if one fails we want the others to continue. The lack of a PR will make it clear that it has failed, and with this new approach the logs should be accessible.